pkg/object/function/storage/storage.go

```diff
@@ -0,0 +1,177 @@
+/*
+ * Copyright (c) 2017, MegaEase
+ * All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package storage
+
+import (
+	"fmt"
+	"time"
+
+	"go.etcd.io/etcd/api/v3/mvccpb"
+
+	"github.com/megaease/easegress/pkg/cluster"
+	"github.com/megaease/easegress/pkg/logger"
+)
+
+const (
+	allFunctionsSpecPrefix  = "/faas/%s/spec"      // +controllerName
+	allFunctionStatusPrefix = "/faas/%s/status"    // +controllerName
+	functionSpecPrefix      = "/faas/%s/spec/%s"   // +controllerName + functionName
+	functionStatusPrefix    = "/faas/%s/status/%s" // +controllerName + functionName
+)
+
+type (
+	// Storage is the interface to contain storage APIs.
+	Storage interface {
```

Comment:
I can see the reason why introduce the Storage.

Reply:
This `Storage` interface is introduced for keeping the controller implementation consistent with `Mesh Controller`.
https://github.com/megaease/easegress/blob/main/pkg/object/meshcontroller/storage/storage.go